### ISSUE TITLE
TypeError when calling Project.__str__ in Python 3

### ISSUE THREAD

**Issue by ghing** [datadesk/python-documentcloud#121]
tl;dr The current implementation of `__str__` returns an 8-bit string in Python 3 (bytes) when it should be a unicode string.

To replicate (I'm running Python 3.6.4):

```
>>> import os
>>> from documentcloud import DocumentCloud
>>> client = DocumentCloud(os.environ.get('DOCUMENTCLOUD_USERNAME'), os.environ.get('DOCUMENTCLOUD_PASSWORD'))
>>> project = client.projects.get(title="Some Project")
>>> print(project)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __str__ returned non-string (type bytes)
```

The problem is with the implementation of [BaseAPIObject.__str__](https://github.com/datadesk/python-documentcloud/blob/master/documentcloud/__init__.py#L570):

```
 def __str__(self):
     return self.__unicode__().encode("utf-8")
```

This implementation is correct for Python 2, but will return a bytes object, not a string in Python 3.

I think what needs to happen is to follow a pattern similar to the one in the [Porting Python 2 Code to Python 3](https://docs.python.org/3.3/howto/pyporting.html#str-unicode) guide:

```
  if sys.version_info[0] >= 3: # Python 3
      def __str__(self):
          return self.__unicode__()
  else:  # Python 2
      def __str__(self):
          return self.__unicode__().encode('utf8')
```

Note that in Python 3, `__unicode__` will never get called directly.

So in Python 3, `__str__` should return a unicode string (the `str` type), which `__unicode__` already does. In Python 2, you have to call encode on the `unicode` object to convert it to an 8-bit string.

**Comment by ghing** [comment]
I'm happy to create a pull request for this, but wanted to make a unit test for this first and was a little unclear about the tests (they mostly failed when I tried to run them in a local checkout).

Are  `DOCUMENTCLOUD_TEST_USERNAME` and `DOCUMENTCLOUD_TEST_PASSWORD` credentials to a specific test account, or should any DocumentCloud credentials work?

**Comment by palewire** [comment]
Please send me a PR. Thanks for spotting it. 

Any credentials should work.

**Comment by ghing** [comment]
Also, where do the slugs for the `version_to_slug` dicts in the tests come from? E.g. https://github.com/datadesk/python-documentcloud/blob/master/tests/test_all.py#L95

**Comment by palewire** [comment]
I don't remember but they are probably documents I uploaded strictly to use in testing. Do you not have access to them?

**Comment by ghing** [comment]
I don't know if I have access or not. The more immediate issue is that there's not entries for 3.6 or 3.7.

**Comment by palewire** [comment]
I think you can just toss in the same slug for those.

**Comment by ghing** [comment]
I fixed this in a776d390a912a61d88ad57e5f6746d8e5ccaa1a6. Closing this issue.
